app/components/AccountSetting/EmailSetting/EmailInput.js

```diff
@@ -27,7 +27,7 @@ export default class EmailInput extends Component {
         </div>
         <div className='input-group text-right'>
           <div className='indicator radius5 font22 text-center email-input'>
-            <input className='email--input text-center' defaultValue={ email } onBlur={ this.handleBlur } />
+            <input className='email--input text-center' onBlur={ this.handleBlur } onChange= { this.props.onEmailChange } value={ currentEmail } />
```

Comment:
I pretty sure that you can put the save email logic on `handleBlur`. Also you won't need the clear flag because the email will remain whether user submit it or not.